src/validation/validationHealthcareProfessional.ts

```diff
@@ -145,3 +199,105 @@ export default function validateNames(
         }
     })
 }
+
+export function validateDegrees(
+    degrees: gqlTypes.InputMaybe<gqlTypes.Degree[]> | undefined,
+    validationResults: Result<unknown>
+): void {
+    if (!degrees || !degrees.length) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'degrees',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+        return
+    }
+
+    if (degrees && degrees.length > 64) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'degrees',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    degrees.forEach((degree, index) => {
+        if (degree.nameEn.length > 64) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `degrees[${index}].nameEn`,
+                errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+                httpStatus: 400
+            })
+        }
+
+        if (degree.nameJa.length > 64) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `degrees[${index}].nameJa`,
+                errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+                httpStatus: 400
+            })
+        }
+
+        if (degree.abbreviation.length > 15) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `degrees[${index}].abbreviation`,
+                errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+                httpStatus: 400
+            })
+        }
+        
+        if (hasJapaneseCharacters(degree.nameEn)) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `degrees[${index}].nameEn`,
+                errorCode: ErrorCode.CONTAINS_JAPANESE_CHARACTER,
+                httpStatus: 400
+            })
+        }
+
+        if (hasLatinCharacters(degree.nameJa)) {
```

Comment:
I don't have this kind of check for Abbreviation. Should abbreviations always be Latin characters?

Reply:
good question! I'm not sure honestly. let's start with latin only for now.